spec/features/idv/doc_auth/document_capture_step_spec.rb

```diff
@@ -41,38 +40,54 @@
     end
   end
 
-  it 'throttles calls to acuant', allow_browser_log: true do
-    allow(IdentityConfig.store).to receive(:doc_auth_max_attempts).and_return(max_attempts)
-    DocAuth::Mock::DocAuthMockClient.mock_response!(
-      method: :post_front_image,
-      response: DocAuth::Response.new(
-        success: false,
-        errors: { network: I18n.t('doc_auth.errors.general.network_error') },
-      ),
-    )
+  context 'throttles calls to acuant', allow_browser_log: true do
+    let(:message) { '' }
+    let(:fake_attempts_tracker) { IrsAttemptsApiTrackingHelper::FakeAttemptsTracker.new }
+    before do
+      allow_any_instance_of(ApplicationController).to receive(
+        :irs_attempts_api_tracker,
+      ).and_return(fake_attempts_tracker)
+      allow(fake_attempts_tracker).to receive(:idv_document_upload_rate_limited)
+      allow(IdentityConfig.store).to receive(:doc_auth_max_attempts).and_return(max_attempts)
+      DocAuth::Mock::DocAuthMockClient.mock_response!(
+        method: :post_front_image,
+        response: DocAuth::Response.new(
+          success: false,
+          errors: { network: I18n.t('doc_auth.errors.general.network_error') },
+        ),
+      )
 
-    allow(IdentityConfig.store).to receive(:doc_auth_max_attempts).and_return(max_attempts)
-    (max_attempts - 1).times do
-      attach_and_submit_images
-      click_on t('idv.failure.button.warning')
+      (max_attempts - 1).times do
+        attach_and_submit_images
+        click_on t('idv.failure.button.warning')
+      end
+
+      freeze_time do
+        attach_and_submit_images
+        timeout = distance_of_time_in_words(
+          Throttle.attempt_window_in_minutes(:idv_doc_auth).minutes,
+        )
+        # rubocop:disable Lint/UselessAssignment
```

Comment:
Could I get a quick explanation of why this disable is needed and the purpose of the assignment?

Reply:
Yeah, this is tricky and I'm open to a better way. Rubocop is incorrectly tagging this as a "useless" assignment, when in fact there is a let for `message` scoped near the top of the test. We need to have the message string created -- with the appropriate timeout value -- before we execute the test assertions. We have to use `freeze_time` here so that we can get the correct timeout value (from the Throttle) for composing the correct expected message.